src/catchup/CatchupWork.cpp

```diff
@@ -343,9 +343,19 @@ CatchupWork::runCatchupStep()
                 seq.push_back(mTransactionsVerifyApplySeq);
             }
 
-            // Step 4.4: Apply buffered ledgers
+            auto onSuccessCallback = [&]() {
```

Comment:
Because we're passing this callback to another class, safest is to only capture things you need, and avoid potential dangling references. Here you can capture ProgressState and catchup mode by value (they remain constant anyway), and have the callback take `ledgerSeq` argument instead of `mLastApplied` (you can deduce last applied ledger by asking LedgerManager). The only thing you need to capture by reference is the progress handler function.

Reply:
fixed